distributed/tests/test_profile.py

```diff
@@ -45,6 +46,47 @@ def test_f():
     assert g['count'] < h['count']
     assert 95 < g['count'] + h['count'] <= 100
 
+    pd = plot_data(state)
+    assert len(set(map(len, pd.values()))) == 1  # all same length
+    assert len(set(pd['color'])) > 1  # different colors
+
+
+@pytest.mark.skipif(
+        WINDOWS,
+        reason='no low-level profiler support for Windows available')
+def test_basic_low_level():
+    pytest.importorskip('stacktrace')
+
+    def test_g():
+        time.sleep(0.05)
+
+    def test_f():
+        for i in range(100):
+            test_g()
+
+    thread = threading.Thread(target=test_f)
+    thread.daemon = True
+    thread.start()
+
+    state = create()
+
+    for i in range(100):
+        frame = sys._current_frames()[thread.ident]
+        llframes = {thread.ident: ll_get_stack(thread.ident)}
+        for f in llframes.values():
+            if f is not None:
+                llprocess(f, None, state)
+
+    assert state['count'] == 100
```

Comment:
This test always fails for me on my OSX machine at this point.  I'm not sure why.

Is there some case where the count would be quite a bit lower than 100?

Reply:
I removed a sleep call from the for loop that processes the stack, which was wrong of me doing.

However, that still leaves another problem, we're relying on sleep for predetermined times to ensure that the the async thread has completed an iteration, this seems to work fine for the python profiler, but not for the low-level one. My guess is that it takes a little longer to process the low-level stack, therefore causing not catching all the calls.

So, what I did now was to remove the async thread completely and rely only on the main thread's sleep, that works for me on Linux and OSX. Regardless, using sleep as a means of thread synchronization is not reliable and we should avoid doing this.